catalog/app/containers/NavBar/BucketControls.js

```diff
@@ -0,0 +1,375 @@
+      setValue(normalizeBucket(evt.target.value));
+    },
+    handleFocus: ({ setAnchor }) => (evt) => {
+      setAnchor(evt.target);
+    },
+    handleBlur: ({ setAnchor, cancel }) => () => {
+      setAnchor(null);
+      if (cancel) cancel();
+    },
+  }),
+  RC.withHandlers({
+    handleKey: ({
+      anchor,
+      go,
+      value,
+      nextSuggestion,
+      prevSuggestion,
+    }) => (evt) => {
+      // eslint-disable-next-line default-case
+      switch (evt.key) {
+        case 'Enter':
+          go(value);
+          break;
+        case 'Escape':
+          if (anchor) anchor.blur();
+          break;
+        case 'ArrowUp':
+          prevSuggestion();
+          break;
+        case 'ArrowDown':
+        case 'Tab':
+          // prevent Tab from switching focus
+          evt.preventDefault();
+          nextSuggestion();
+          break;
+      }
+    },
+    handleSuggestion: ({ go, setValue }) => (s) => {
+      setValue(s);
+      go(s);
+    },
+  }),
+  withStyles(({ spacing: { unit }, zIndex }) => ({
+    input: {
+      marginLeft: unit * 2,
+    },
+    popper: {
+      zIndex: zIndex.appBar + 1,
+    },
+    item: {
+      paddingBottom: 20,
+      paddingTop: 20,
+    },
+    description: {
+      maxWidth: 50 * unit,
+    },
+    icon: {
+      height: 40,
+      width: 40,
+    },
+    help: {
+      paddingLeft: 2 * unit,
+      paddingTop: unit,
+    },
+  })),
+  ({
+    anchor,
+    autoFocus = false,
+    value,
+    handleKey,
+    handleChange,
+    buckets,
+    handleSuggestion,
+    handleFocus,
+    handleBlur,
+    classes,
+  }) => (
+    <React.Fragment>
+      <NavInput
+        startAdornment={<InputAdornment>s3://</InputAdornment>}
+        value={value}
+        className={classes.input}
+        autoFocus={autoFocus}
+        onKeyDown={handleKey}
+        onChange={handleChange}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        placeholder=" Enter bucket name"
+      />
+      <Popper
+        open={!!anchor}
+        anchorEl={anchor}
+        placement="bottom-end"
+        className={classes.popper}
+      >
+        <Paper>
+          <MenuList>
+            {buckets.map((s) => (
+              <MenuItem
+                className={classes.item}
+                key={s.name}
+                onMouseDown={() => handleSuggestion(s.name)}
+                selected={s.name === value}
+              >
+                <img src={s.icon} alt={s.title} className={classes.icon} />
+                <ListItemText
+                  primary={s.title}
+                  secondary={s.description}
+                  secondaryTypographyProps={{
+                    noWrap: true,
+                    className: classes.description,
+                  }}
+                  title={s.description}
+                />
+              </MenuItem>
+            ))}
+            <li className={classes.help}>
+              <a href="TODO">Learn how to create your own registry</a>
```

Comment:
@ResidentMario where to?

Reply:
https://github.com/quiltdata/t4/tree/master/deployment#installation

Reply:
```suggestion
              <a href="https://github.com/quiltdata/t4/tree/master/deployment#installation">
                Learn how to create your own registry
              </a>
```